### ISSUE TITLE
feat(semver): support versions which satisfy a range

### ISSUE THREAD

**Issue by dylanaubrey** [JamieMason/syncpack#128]
## Description

I have peer dependencies declared using `<` and dependencies/devDependencies declareed with `^`. Using the first entry in the below terminal output snippet as an example, for `@babel/cli` `^7.19.3` and `<8` are equivalent, but list-mismatches is flagging that up as an error. Even if I change to `^7.19.3` and `^7`, which is also equivalent and uses the same semver range operator, list-mismatches is still flagging as an error. I expected list-mismatches to be able to handle equivalent semver ranges.

```sh
✘ @babel/cli <8 is the highest valid semver version in use
  ^7.19.3 in devDependencies of package.json
✘ @babel/core <8 is the highest valid semver version in use
  ^7.20.5 in devDependencies of package.json
✘ @babel/eslint-parser <8 is the highest valid semver version in use
  ^7.19.1 in devDependencies of package.json
✘ @babel/plugin-proposal-class-properties <8 is the highest valid semver version in use
  ^7.18.6 in devDependencies of package.json
```

## Suggested Solution

Without going through the code, it seems just a string equality comparison is being done, rather than comparing the two ranges using something like the `semver` package and `semver.satisfies`.

## Help Needed

The function `getUniqueVersions` is using a util that is reducing the instance versions down to unique entries and then `hasMismatchingVersions` is checking if those entries are greater than one. https://github.com/JamieMason/syncpack/blob/0192cb56061631634caf8f3f3427ada9a92cf439/src/get-context/get-groups/version-group/instance-group/index.ts#L50

I would say this should be swapped out for something more advanced with potentially a configuration option to drive its behaviour. Maybe a default "exactMatch" option could be "as is" so no breaking change, a "satisfies" option that checks all versions are within the same range, and a "satisfiesPeers" option that does exactMatch checks among all depencency/devDependency versions and among all peerDependency versions, but a satisfies check of all depencency versions against peerDependency versions. 

This needs a bit more thought, but it would be good to hear what you think as the scenario of declaring dependency/devDependency semver ranges differently to peerDependency semver ranges is a common one and it would be good if this library supported that.

The latter two options are more intensive checks, but they are opt ins so the users would be opting in with the knowledge of this.

**Comment by JamieMason** [comment]
Thanks a lot @dylanaubrey, it'll be a pretty big job but I like your idea of how to go about it. Will add this on the backlog.

**Comment by JamieMason** [comment]
I've defined the config and written the docs for this in [branches/satisfies](https://github.com/JamieMason/syncpack/tree/satisfies), but not started implementing it yet. If anyone has any different preferences for how this should look, let me know:

<img width="1440" alt="Screenshot 2023-06-03 at 12 30 01" src="https://github.com/JamieMason/syncpack/assets/320492/819730d2-ade4-4d40-b41d-e6e95d05bbe3">

**Comment by JamieMason** [comment]
Released in [10.5.1](https://github.com/JamieMason/syncpack/releases/tag/10.5.1) which:

- Adds a new [`syncpack prompt`](https://jamiemason.github.io/syncpack/prompt) command which displays a series of prompts to fix version mismatches which syncpack cannot fix automatically.
- Adds a new [Same Range Version Group](https://jamiemason.github.io/syncpack/config/version-groups/same-range) which relaxes syncpack to ensure that all versions have semver ranges which all satisfy each other, instead of the default behaviour of them having to be identical.

See the full [release notes](https://github.com/JamieMason/syncpack/releases/edit/10.5.1).

Any problems, please let me know. 

If you find syncpack useful, please star the project or help us spread the word on social media.
